frontend/src/components/PostRecipe.jsx

```diff
@@ -154,155 +177,263 @@ const RecipePost = () => {
-                            </div>
-    
-                            <div className="preparation-section">
-                                <label>PREPARATION</label>
-                                <div className="preparation-list">
-                                    {preparation.map((step, index) => (
-                                        <div key={index} className="preparation-item">
-                                            <span>Step {step.step_number}: {step.body}</span>
-                                            <button id='buttons_postRecipe' onClick={() => handleInstructionDelete(index)}>X</button>
+                                    </Form.Select>
+                                    </Col>
+                                    </Row>
+                                    <div className="d-grid gap-2">
+                                    <Button className='mb-3' onClick={addIngredientField} variant="danger">Add ingredient</Button>
+                                    </div>
+                                    
+                        </Col>
+                        <Col xs={6} md={6} lg={6}>
+                            <label>PREPARATION</label>
+                                    <Row className='mt-2'>
+                                        <Col sm={12}>
+                                            <Form.Control type="text"
+                                            ref={instructionRef}
+                                            placeholder={`Step ${preparation.length > 0 ? preparation.length + 1 : 1}`}
+                                            maxLength={2000}
+                                            />
+                                        </Col>
+                                        <Col sm={12}>
+                                        <div className="d-grid gap-2">
+                                        <Button className='mb-3' style={{marginTop: '10px'}} onClick={addInstructionField} variant="danger">Add step</Button>{' '}
                                         </div>
-                                    ))}
-                                </div>
-                                <div className="instruction-input-group">
-                                    <textarea 
-                                        id='textArea_postRecipe' 
-                                        ref={instructionRef}
-                                        placeholder={`Step ${preparation.length > 0 ? preparation.length + 1 : 1}`}
-                                        maxLength={2000}
-                                    />
-                                </div>
-                                <button id='buttons_postRecipe' style={{marginTop: '10px'}} onClick={addInstructionField}>Add step</button>
-                            </div>
-                        </div>
-    
-                        <div className="recipe-details">
-                            <UploadFile myParentCallback={handleCallback} />
```

Comment:
Instead of directly uploading the big raw images to the Mongo Database, which is using a free version and has only 512MB storage, it's better to use an external Database, for example Google Cloud Storage buckets. In the previous version I was using my own component UploadFile which did that function. Probably you could still use this component and maybe style it there as a component for if we need to upload more files elsewhere.

The UploadFile component basically made a POST to http://127.0.0.1:8000/recipe/uploadfile with the image, which uploads the file to Google Cloud Storage and returns the URL.

To be able to work you would need to authenticate to Google Cloud, which is done with putting a private file in the backend, which is explained in the Discord.

To ease the changes I can later edit it myself and explain you how can you put this key.

Reply:
Okay, thanks Andreu. It will be easy for future feature implementations.